src/managers/image-generator.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+const TextToImage = require('text-to-image');
+const Jimp = require('jimp');
+const FONT_SIZE_TO_LINE_HEIGHT = 1.1;
+const JPEG_QUALITY = 80;
+const { registerFont } = require('canvas');
+
+
+/**
+ * Image generator
+ * This class allows to generate new image based on baseImage using a template rules
+ */
+class ImageGenerator {
+
+  /**
+   * Register assets
+   */
+  constructor() {
+    registerFont('assets/fonts/MyriadPro-Bold.otf', { family: 'Myriad Pro' });
+  }
+
+  async generateImage(picUrl, params, templateConfig) {
+    const baseImg = await Jimp.read(picUrl);
+    await this.prepareBaseImage(baseImg, params, templateConfig);
+    await this._composeImage(baseImg, templateConfig.items, params);
+    return baseImg;
+  }
+
+  /**
+   * Prepare base image for according to template requirements
+   * @param baseImg
+   * @param params
+   * @param templateConfig
+   * @returns {Promise<void>}
+   */
+  async prepareBaseImage(baseImg, params, templateConfig) {
+    await baseImg.quality(JPEG_QUALITY);
+
+    if (templateConfig.input) {
+      if (templateConfig.input.type === 'fixed') {
+        const basePicRatio = baseImg.getWidth() / baseImg.getHeight();
+        const inputRatio = templateConfig.input.width / templateConfig.input.height;
+        if (basePicRatio <= inputRatio) {
+          await baseImg.scaleToFit(templateConfig.input.width, Jimp.AUTO);
+        } else {
+          await baseImg.scaleToFit(Jimp.AUTO, templateConfig.input.height);
+        }
+        let x = (baseImg.getWidth() - templateConfig.input.width) / 2 + params.xShift || 0;
+        let y = (baseImg.getHeight() - templateConfig.input.height) / 2 + params.yShift || 0;
+        x = x < 0 ? 0 : x;
+        y = y < 0 ? 0 : y;
+        await baseImg.crop(x, y, templateConfig.input.width, templateConfig.input.height);
+      }
+    }
+  }
+
+  /**
+   * Compose baseImage using template rules
+   * @param baseImg
+   * @param itemConfigList
+   * @param params
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _composeImage(baseImg, itemConfigList, params) {
```

Comment:
Is there a reason of using the underscore '_' in the beginning of the method's name? Just asking, in case it's feasible to use this as a naming convention in other places.

If it's because of the 'private' modifier, then why "prepareBaseImage" method is not private?